### PR TITLE
New assets tab

### PR DESCRIPTION
**Pull request by nuurcodes**
Screenshot 2021-05-29 at 02 28 05

### CODE REVIEW

File: src/components/crucible/detail/assets.tsx [hunk 1]
```diff
@@ -1,48 +1,49 @@
-// TEST PAGE
-import { useEffect } from 'react';
-import { useCrucibles } from '../../../store/crucibles';
-import { Box, VStack } from '@chakra-ui/layout';
-import { StatGroup } from '@chakra-ui/stat';
-import { Button } from '@chakra-ui/button';
-import StatCard from '../../shared/StatCard';
+import React, { useEffect, useMemo } from 'react';
+import { Crucible, useCrucibles } from '../../../store/crucibles';
```
Comment (georgeweiler) [inline src/components/crucible/detail/assets.tsx:2]: do we have an eslint rule configured to put relative imports below absolute module imports?

File: src/components/crucible/detail/assets.tsx [hunk 2]
```diff
@@ -1,48 +1,49 @@
...
 const Assets = () => {
-  const {
-    crucibles,
-    cruciblesLoading,
-    getOwnedCrucibles,
-    resetCrucibles,
-  } = useCrucibles();
+  const history = useHistory();
+  const { account } = useWeb3React();
+  const { crucibleId } = useParams<{ crucibleId: string }>();
```
Comment (georgeweiler) [inline src/components/crucible/detail/assets.tsx:12]: I didn't know about this hook! super cool

File: src/components/crucible/detail/assets.tsx [hunk 3]
```diff
@@ -1,48 +1,49 @@
...
+  }, [crucibles, crucibleId]);
+
   if (cruciblesLoading) {
-    return <div>Loading...</div>;
+    return <AssetsLoadingSkeleton />;
   }
 
   if (crucibles.length === 0) {
-    return <div>No crucibles</div>;
+    history.push('/');
```
Comment (georgeweiler) [inline src/components/crucible/detail/assets.tsx:29]: We can return `<Redirect to="/">` here

File: src/components/crucible/detail/assets-detail/transferErc20.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+  const { library } = useWeb3React();
+  const { invokeContract, ui } = useContract(
+    type === 'withdraw' ? withdrawFromCrucible : depositToCrucible
+  );
+  const [isMax, setIsMax] = useState(false);
+  const [amount, setAmount] = useState('0');
+
+  if (error) {
+    return <Text>{error}</Text>;
```
Comment (georgeweiler) [inline src/components/crucible/detail/assets-detail/transferErc20.tsx:57]: How does the UI for this error look? Should we wrap in a Box with the same padding and styles as the happy path component?

File: src/helpers/crucible.ts [hunk 5]
```diff
@@ -53,53 +54,45 @@ export const getContainedAssets = async (
...
-          tokenName: tx.tokenName,
-          tokenSymbol: tx.tokenSymbol,
-          value:
-            tx.from === address
-              ? BigNumber.from(tx.value).mul(-1)
-              : BigNumber.from(tx.value),
-        };
-      }
-      return tx;
+    data.result.forEach((tx) => {
```
Comment (georgeweiler) [inline src/helpers/crucible.ts:64]: I suspect there's a way to use `Array.reduce` to clean this helper up,  but it's not a huge deal.

File: src/hooks/useContainedAssets.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  const handleSetSelectedAsset = (address: string) => {
+    setSelectedAsset(
+      containedAssets.find((asset) => asset.contractAddress === address) ||
+        containedAssets[0]
+    );
+  };
+
+  useEffect(() => {
+    setContainedAssetsLoading(true);
```
Comment (georgeweiler) [inline src/hooks/useContainedAssets.tsx:89]: let's put this loading flag setter in the async function itself so that other callers will also flip th flag.

**Review by georgeweiler — APPROVED** [review]
Approving with a few comments for consideration. Great job @nuurcodes

File: src/components/crucible/detail/assets.tsx (hunk 1 above)
Comment (nuurcodes) [reply src/components/crucible/detail/assets.tsx:2]: No, do you think it's worth having one?

File: src/components/crucible/detail/assets-detail/transferErc20.tsx (hunk 4 above)
Comment (nuurcodes) [reply src/components/crucible/detail/assets-detail/transferErc20.tsx:57]: Good suggestion!

File: src/components/crucible/detail/assets.tsx (hunk 3 above)
Comment (nuurcodes) [reply src/components/crucible/detail/assets.tsx:29]: Great suggestion

**Pull request merged**